src/main/java/org/opensearch/ad/stats/suppliers/ModelsOnNodeCountSupplier.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.ad.stats.suppliers;
+
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import org.opensearch.ad.caching.CacheProvider;
+import org.opensearch.ad.ml.ModelManager;
+
+/**
+ * ModelsOnNodeSupplier provides a List of ModelStates info for the models the nodes contains
+ */
```

Comment:
Seems this class only returns the count of models instead of a List of ModelStates?

Reply:
Good catch.  fixed.